pkg/api/login.go

```diff
@@ -159,8 +161,16 @@ func (hs *HTTPServer) LoginAPIPing(c *models.ReqContext) Response {
 }
 
 func (hs *HTTPServer) LoginPost(c *models.ReqContext, cmd dtos.LoginCommand) Response {
+	action := "login"
 	if setting.DisableLoginForm {
-		return Error(401, "Login is disabled", nil)
+		errMsg := "Login is disabled"
+		hs.SendLoginLog(&models.SendLoginLogCommand{
+			ReqContext: c,
+			LogAction:  action,
+			HTTPStatus: http.StatusUnauthorized,
+			Error:      errors.New(errMsg),
+		})
```

Comment:
Would it make sense to use defer to send this rather than defining it at all the exit points? 🤔

If we could pick up the status and error from the response object being returned instead of having to duplicate it everywhere, that'd reduce the complexity of the rest of the logic in this function quite a bit.

Reply:
Good point! I searched for a better way of doing this but didn't think about defer.